lib/rubocop/cop/chef/modernize/includes_mixin_shellout.rb

```diff
@@ -42,13 +42,35 @@ class IncludingMixinShelloutInResources < Cop
             (send nil? :require ( str {"chef/mixin/shell_out" "chef/mixin/powershell_out"} ))
           PATTERN
 
+          def_node_search :hwrp_classes?, <<-PATTERN
+            (class
+              (const ... )
+              {(const
+                (const
+                  (const nil? :Chef) :Provider) :LWRPBase)
+               (const
+                 (const nil? :Chef) :Provider)
+               }
+              ...)
+          PATTERN
+
+          def check_for_offenses(node)
+            containing_dir = File.basename(File.dirname(processed_source.path))
+
+            if containing_dir == 'resources' # resource
+              add_offense(node, location: :expression, message: MSG, severity: :refactor)
+            elsif hwrp_classes?(processed_source.ast) # hwrp
+              add_offense(node, location: :expression, message: MSG, severity: :refactor)
+            end
```

Comment:
Are both branches of this conditional supposed to be the same? If so, could you combine them?

Reply:
Yeah at one point last night there was reason for this, but that reason is way gone.